virtual_rainforest/models/abiotic/radiation.py

```diff
@@ -0,0 +1,215 @@
+            surface_temperature: surface soil temperature [C]
+        """
+        # longwave emission canopy
+        self.longwave_canopy = (
+            CANOPY_EMISSIVITY
+            * BOLZMAN_CONSTANT
+            * (CELSIUS_TO_KELVIN + canopy_temperature) ** 4
+        )
+
+        # longwave emission surface
+        self.longwave_soil = (
+            SOIL_EMISSIVITY
+            * BOLZMAN_CONSTANT
+            * (CELSIUS_TO_KELVIN + surface_temperature) ** 4
+        )
+
+    def calc_netradiation_surface(
+        self,
+        canopy_absorption: NDArray[np.float32],
+    ) -> None:
+        """Calculates daily net radiation at the forest floor [J m-2].
+
+        Args:
+            canopy_absorption: absorption by canopy layers, [J m-2]
+        """
+        self.netradiation_surface = (
+            self.topofcanopy_radiation
+            - canopy_absorption  # np.sum(canopy_absorption, axis=1)  # what axis?
+            - self.longwave_soil
+            - self.longwave_canopy  # np.sum(self.longwave_canopy, axis=1)
+        )
+
+    def radiation_balance(
+        self,
+        elevation: NDArray[np.float32],
+        shortwave_in: NDArray[np.float32],
+        sunshine_hours: NDArray[np.float32],
+        albedo_vis: NDArray[np.float32],
+        albedo_shortwave: NDArray[np.float32],
+        canopy_temperature: NDArray[np.float32],
+        surface_temperature: NDArray[np.float32],
+        canopy_absorption: NDArray[np.float32],
+    ) -> None:
+        """Calculate radiation balance.
+
+        The function runs the full radiation balance and populates the following
+        attributes:
+            * elevation [m]
+            * top of canopy downward shortwave radiation, [J m-2]
+            * top of canopy photosynthetic photon flux density,[mol m-2]
+            * longwave radiation from n individual canopy layers [J m-2]
+            * longwave radiation from soil [J m-2]
+            * net shortwave radiation at the surface (=forest floor)[J m-2]
+
+        Args:
+            elevation: elevation [m]
+            shortwave_in: daily downward shortwave radiation[J m-2]
+            sunshine_hours: fraction of sunshine hours, between 0
+                (100% cloud cover) and 1 (cloud free sky)
+            albedo_vis: visible light albedo, default = 0.03
+            albedo_shortwave: shortwave albedo, default = 0.17
+            canopy_temperature: canopy temperature of n layers [C]
+            surface_temperature: surface soil temperature [C]
+            canopy_absorption: absorption by canopy [J m-2]
+        """
+
+        tau = self.calc_ppfd(shortwave_in, sunshine_hours, albedo_vis)
+        self.calc_topofcanopy_radiation(
+            tau, shortwave_in, sunshine_hours, albedo_shortwave
+        )
+        self.calc_longwave_radiation(canopy_temperature, surface_temperature)
+        self.calc_netradiation_surface(canopy_absorption)
+        del tau
```

Comment:
Think this was mentioned on a different pull request, but explicit deletion of data is not often necessary/useful in python (this is in contrast to some other languages)

Reply:
They will be garbage collected when the function exits (see #150)